db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -3140,7 +3151,8 @@ Status DBImpl::BackgroundCompaction(bool* made_progress,
         c->mutable_cf_options()->paranoid_file_checks,
         c->mutable_cf_options()->report_bg_io_stats, dbname_,
         &compaction_job_stats, thread_pri, io_tracer_,
-        is_manual ? &manual_compaction_paused_ : nullptr, db_id_,
+        is_manual ? &manual_compaction_paused_ : nullptr,
+        is_manual ? manual_compaction->canceled : nullptr, db_id_,
```

Comment:
I don't have an opinion yet on whether consolidation is worthwhile, but it does feel we have three checks (`shutting_down_`, `manual_compaction_paused_`, `manual_compaction_canceled_`) that could be reduced.

One idea (not fully thought through) to eliminate  `manual_compaction_paused_`: we could make the `std::atomic<bool>*` in `ManualCompactionState` point to something managed by us when not provided by user. Then in `DisableManualCompaction()` we would iterate through the `ManualCompactionState`s in `manual_compaction_dequeue_` and set each of their `*canceled = true`

Reply:
I considered this, but worried about the multi-threaded implications.

Currently `DisableManualCompaction()` atomically shuts down all compactions, if we were to set a series of variables, then we wouldn't be atomically disabling all compactions without adding additional locking, which would be more nuanced and could potentially break some use-cases.  Without a deeper understanding of how the system works and how `DisableManualCompaction` was used, I didn't feel fully comfortable merging the two behaviors together.